src/border_agent/border_agent.cpp

```diff
@@ -252,8 +251,14 @@ void BorderAgent::PublishMeshCopService(void)
         }
         else
         {
-            activeTimestamp = htobe64(activeDataset.mActiveTimestamp);
-            txtList.emplace_back("at", reinterpret_cast<uint8_t *>(&activeTimestamp), sizeof(activeTimestamp));
+            uint64_t activeTimestampValue;
+
+            activeTimestampValue = (activeDataset.mActiveTimestamp.mSeconds << 16) |
```

Comment:
nit: define constant for 16 and 1.

Reply:
@wgtdkp , thoughts?

Reply:
I think it's fine to use the numbers here. It should be easy to understand.

Reply:
I tend to not define constants too far away from where they are being used, especially when there are a single use case. To help readability, I made a short function for converting the timestamp to uint64_t and added documentation for its layout.